index/index.go

```diff
@@ -337,34 +361,64 @@ func (idx *Indexer) loopIteration(
 }
 
 func (idx *Indexer) ensureIndex(
-	ctx context.Context, indexType string, docType string,
+	ctx context.Context, indexType string, docType string, lang string,
 ) (string, error) {
-	name := fmt.Sprintf("%s-%s-%s",
-		indexType, idx.name, nonAlphaNum.ReplaceAllString(docType, "_"))
+	safeDocType := nonAlphaNum.ReplaceAllString(docType, "_")
+	settings := GetIndexSettings(lang)
+
+	index := fmt.Sprintf("%s-%s-%s-%s", indexType, idx.name, safeDocType, settings.Name)
+
+	typeAlias := fmt.Sprintf("%s-%s-%s", indexType, idx.name, safeDocType)
+	langAlias := fmt.Sprintf("%s-%s-%s-%s", indexType, idx.name, safeDocType, settings.Language)
 
-	existRes, err := idx.client.Indices.Exists([]string{name},
+	existRes, err := idx.client.Indices.Exists([]string{index},
 		idx.client.Indices.Exists.WithContext(ctx))
 	if err != nil {
 		return "", fmt.Errorf("check if index exists: %w", err)
 	}
 
-	if existRes.StatusCode == http.StatusOK {
-		return name, nil
+	if existRes.StatusCode != http.StatusOK {
+		res, err := idx.client.Indices.Create(index,
+			idx.client.Indices.Create.WithBody(strings.NewReader(settings.Settings)),
+			idx.client.Indices.Create.WithContext(ctx))
+		if err != nil {
+			return "", fmt.Errorf("create index %q: %w", index, err)
+		}
+
+		defer res.Body.Close()
+
+		if res.StatusCode != http.StatusOK {
+			return "", fmt.Errorf("server response: %s", res.Status())
+		}
 	}
 
-	res, err := idx.client.Indices.Create(name,
-		idx.client.Indices.Create.WithContext(ctx))
-	if err != nil {
-		return "", fmt.Errorf("create index %q: %w", name, err)
+	if typeAlias != "" {
```

Comment:
That's correct. This check is mostly for aesthetic reasons since we do it for the other alias. I guess o nicer way to do it would be to move the check inside the `ensureAlias` func.

Reply:
The original way this was handled was to put all unidentified documents into the `standard` index, but that wasn't especially elegant. I've reworked the language settings functionality now, so even a document with a language we don't know how to handle (like `ja-JP`) will go into the correct index and get the correct language alias. Thus no need to check for empty alias names anymore.